tools/cache.py

```diff
@@ -41,30 +31,31 @@ def acquire_cache_lock(self):
       # should never happen
       raise Exception('Attempt to lock the cache but FROZEN_CACHE is set')
 
-    if not self.EM_EXCLUSIVE_CACHE_ACCESS and self.acquired_count == 0:
+    if self.acquired_count == 0:
       logger.debug('PID %s acquiring multiprocess file lock to Emscripten cache at %s' % (str(os.getpid()), self.dirname))
+      # This should never happen because any calls to emcc make when populating the
+      # cache should not themselves attempt to acquire the cache lock.  For example,
+      # it should never be the case that while holding the cache lock to build libraryA
+      # a build of libraryB is needed.  This would be a bug in the dependecy graph
+      # since if libraryA depended on libraryB it should already have been built and
+      # cached before the libraryA build started.
+      assert 'EM_PARENT_HOLD_CACHE_LOCK' not in os.environ
```

Comment:
`EM_PARENT_HOLD_CACHE_LOCK` appears here but below it is `EM_PARENT_HOLDS_CACHE_LOCK` with an `S` in the middle (after `HOLD`)

I guess we don't have tests that can check that - seems hard to test. But maybe its worth testing some stress cases manually?

Reply:
Good point.   This is hard to test from the outside because if you hit this it would be an internal bug/inconsistency in emscripten.     Testing this would probably require some kind of extra tests flags/features or using unit tests (which IIRC we don't yet have in emscripten.   Should we maybe add some ?

Reply:
Maybe, but not urgent I guess.

For now, landing this and waiting a while before a release is hopefully good enough to catch any bugs in `tot` builds.